shared-worker/firebase-worker.js

```diff
@@ -0,0 +1,54 @@
+importScripts('../firebase/firebase-app.js');
+importScripts('../firebase/firebase-firestore.js');
+var config = {
+  // your config here
+};
+
+/** Port used to communicate with the shared worker. */
+self.sharedWorkerPort = null;
+
+/** Whether this Firestore worker holds the lease. */
```

Comment:
What is "the lease"?  I think you're reusing terminology based on implementation details from the Firestore SDK but it won't make any sense to our developers.  Perhaps:

hasPrimaryLease => isUsingPersistence
releaseLease => releasePersistence
leaseAvailable => persistenceAvailable
etc...

The basic idea is the developer needs to keep track of whether the web worker is using persistence and/or can force using persistence in the current web worker (without breaking other tabs / web workers).  The "lease" is how we manage this in the SDK for ourselves, but "lease" probably isn't the best terminology to use outside the SDK (since there are no lease timeouts, etc.)

If you thought it was simpler you could describe it as a "lock", but you'd still want to have some big explanatory comments to explain what we're locking and why, etc. without assuming the developer has any knowledge of Firestore SDK internals.